### ISSUE TITLE
Undefined variable unicode

### ISSUE THREAD

**Issue by jasminabasurita** [raghur/vim-ghost#25]
I get an error any time I try to use this. I have these 2 lines in my vimrc which I don't think should cause an issue with this
```
set encoding=utf-8   " Use Utf-8
scriptencoding utf-8
```

```
error caught while executing async callback:                                                                                                                                                                                                                                      
NvimError(b'Vim(echo):E121: Undefined variable: unicode')                                                                                                                                                                                                                         
Traceback (most recent call last):                                                                                                                                                                                                                                                
  File "/home/user/.vim/plugged/vim-ghost/rplugin/python3/ghost.py", line 191, in _handle_on_message                                                                                                                                                                             
    slugify(req["title"])[:50]                                                                                                                                                                                                                                                    
  File "/usr/lib/python3.7/site-packages/slugify.py", line 24, in slugify                                                                                                                                                                                                         
    unicode(                                                                                                                                                                                                                                                                      
NameError: name 'unicode' is not defined                                                                                                                                                                                                                                          
During handling of the above exception, another exception occurred:                                                                                                                                                                                                               
Traceback (most recent call last):                                                                                                                                                                                                                                                
  File "/home/user/.vim/plugged/vim-ghost/rplugin/python3/ghost.py", line 211, in _handle_on_message                                                                                                                                                                             
    self.nvim.command("echo '%s'" % ex)                                                                                                                                                                                                                                           
  File "/usr/lib/python3.7/site-packages/neovim/api/nvim.py", line 272, in command                                                                                                                                                                                                
    return self.request('nvim_command', string, **kwargs)                                                                                                                                                                                                                         
  File "/usr/lib/python3.7/site-packages/neovim/api/nvim.py", line 170, in request                                                                                                                                                                                                
    res = self._session.request(name, *args, **kwargs)                                                                                                                                                                                                                            
  File "/usr/lib/python3.7/site-packages/neovim/msgpack_rpc/session.py", line 100, in request                                                                                                                                                                                     
    raise self.error_wrapper(err)                                                                                                                                                                                                                                                 
neovim.api.nvim.NvimError: b'Vim(echo):E121: Undefined variable: unicode'                                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                  
the call was requested at                                                                                                                                                                                                                                                         
  File "/usr/lib/python3.7/site-packages/SimpleWebSocketServer/SimpleWebSocketServer.py", line 566, in _parseMessage                                                                                                                                                              
    self._handlePacket()                                                                                                                                                                                                                                                          
  File "/usr/lib/python3.7/site-packages/SimpleWebSocketServer/SimpleWebSocketServer.py", line 238, in _handlePacket                                                                                                                                                              
    self.handleMessage()                                                                                                                                                                                                                                                          
  File "/home/user/.vim/plugged/vim-ghost/rplugin/python3/ghost.py", line 37, in handleMessage                                                                                                                                                                                   
    self.server.context.on_message(req, self)                                                                                                                                                                                                                                     
  File "/home/user/.vim/plugged/vim-ghost/rplugin/python3/ghost.py", line 237, in on_message                                                                                                                                                                                     
    self.nvim.async_call(self._handle_on_message, req, websocket)  
```

**Comment by raghur** [comment]
That's strange...Are you running this with py2 by any chance?

**Comment by stappersg** [comment]
Background information:

- Install was done with `vim-plug`
- Install was trouble some:
-- Did manually ":GhostInstall"   ( as if the the `{'do': ':GhostInstall'}` from `:PlugInstall` wasn't executed )
-- Did manually `sudo pip3 install simple-web-socket-server`
-- Did manually `sudo pip3 install slugify`
- Editor is neovim
- Webbrowser is Mozilla FireFox
- Desktop environment is XFCE

**Comment by raghur** [comment]
> sudo pip3 install slugify
That's not the right module - the correct one is `python-slugify`. Also, you don't need the modules to be installed system wide (user installs are fine) Uninstall that module 

`sudo pip3 uninstall slugify`

and then run from the vim-ghost folder,

`pip3 install --user -r requirements_posix.txt`

`:GhostInstall` basically does that - but when vim-plug calls it, the current working directory is the checkout folder. Probably when you're calling it explicitly, working folder is something else and hence it doesn't find the install script. I'd like to track this down if you are willing to spend some time.

**Comment by stappersg** [comment]
Yes, I do think that I have the latest.
```
gs0604@trancilo:~/.config/nvim/plugged/vim-ghost
$ pwd
/home/gs0604/.config/nvim/plugged/vim-ghost
gs0604@trancilo:~/.config/nvim/plugged/vim-ghost
$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
gs0604@trancilo:~/.config/nvim/plugged/vim-ghost
$ git config --list | grep url
remote.origin.url=https://github.com/raghur/vim-ghost.git
gs0604@trancilo:~/.config/nvim/plugged/vim-ghost
$ git show | head -n 8
commit ded94381d26b233881a61fe55a4d0515af38db5f
Author: Raghu <raghur@users.noreply.github.com>
Date:   Mon Dec 31 16:44:55 2018 +0530

    Fix version check for vim 8

diff --git a/.gitignore b/.gitignore
new file mode 100644
gs0604@trancilo:~/.config/nvim/plugged/vim-ghost
$ 
```

**Comment by dushoff** [comment]
I had a similar problem, but could not be resolved through GhostInstall. 

I think GhostInstall does not work for me because of my Debian python installation. 

I needed to make a python virtual environment, and activate it before running nvim. But GhostInstall still does not work for me in this case. Instead I needed to manually run pip in the virtual environment.

This comment is my first successful use of GhostText! Thanks for your module.
